### PR TITLE
Stricter validation for URLs in Search Dev Tools

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

This PR adds a strict validation for the URL argument for search/dev-tools REST endpoint. 

We used to do a fairly naive check - a valid URL ending in '_search'  would pass it. Theoretically, that can be exploited.

With PR validation is much stricter now.

### Plugin Updated: Search Dev Tools

We've added strict validation rules for the REST endpoint arguments

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1.  Add `$query['url'] = 'somethingbogus'` around https://github.com/Automattic/vip-go-mu-plugins/blob/92f77df56e3c3a3268b9abd080f14b1cedae8680/search/search-dev-tools/search-dev-tools.php#L134
2. Open Dev Tools and try to edit a query
3. The remote request to the REST endpoint should be a 400 now.

### CODE REVIEW

File: tests/search/test-search-dev-tools.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	public function data_provider_endpoint_urls() {
+		return [
+			[
+				'input'    => 'http://vip-search:9200/vip-123-v1/_search',
+				'expected' => true,
+			],
+			[
+				'input'    => 'http://vip-search:9200/vip-123-v1/_search',
```
Comment (nickdaugherty) [inline tests/search/test-search-dev-tools.php:33]: Looks like this input is duplicated - did you mean to use a different endpoint than `_search`?

**Review by nickdaugherty — APPROVED** [review]
Nice, thank you! I left one minor note on a test case, otherwise should be good to go.

File: tests/search/test-search-dev-tools.php (hunk 1 above)
Comment (rinatkhaziev) [reply tests/search/test-search-dev-tools.php:33]: Oh that meant to catch the multiple indexes, another failed copypasta.

Thanks for the catch!

**Comment by rinatkhaziev** [comment]
r1596-stacks

**Pull request merged**